src/Nethermind/Nethermind.Blockchain/Synchronization/FastBlocks/FastBlocksFeed.cs

```diff
@@ -884,11 +884,16 @@ private int InsertHeaders(FastBlocksBatch batch)
             int rightFillerSize = (int) (batch.Headers.EndNumber - addedLast);
             if (added + leftFillerSize + rightFillerSize != batch.Headers.RequestSize)
             {
-                throw new Exception($"Added {added} + left {leftFillerSize} + right {rightFillerSize} != request size {batch.Headers.RequestSize} in  {batch}");
+                throw new Exception($"Added {added} + left {leftFillerSize} + right {rightFillerSize} != request size {batch.Headers.RequestSize} in {batch}");
             }
 
             added = Math.Max(0, added);
 
+            if (added > 0)
+            {
```

Comment:
not flushed on exceptions? again, I think that this flash introduces complexity that is internal to blocktree but somehow leaked outside